### ISSUE TITLE
Option for a JST global instead of ES6?

### ISSUE THREAD

**Issue by searls** [joliss/broccoli-template#3]
Would you be open to an option that would set a `window.JST` global (similar to how sprockets/jammit does it) as opposed to an ES6 module?

**Comment by joliss** [comment]
Possibly. Can we come up with a reasonable API for that? I don't want the plugin to be confusing to use. What are our options in terms of plugin API?

**Comment by searls** [comment]
Thanks for replying, Jo! Apologies as I'm on vacation right now but I was probably naively assuming an option flag specifying a global name (where undefined/null/blank strong would leave the default es6 behavior intact). 

I suspect that for the non-Ember CLI crowd, that'd be necessary for this plugin to be useful.

**Comment by joliss** [comment]
Do you plan on initializing the global, or is that never necessary?

``` js
if (window.JST == null) window.JST = {}
window.JST['foo'] = "..."
```

**Comment by searls** [comment]
I think that would probably be necessary, yeah.

**Comment by joliss** [comment]
So does everyone want to initialize? And do they initialize the same way (using `= {}`)?

**Comment by searls** [comment]
I suspect everyone would want a single initialize and that it should only initialize if the global is not already defined.

**Comment by joliss** [comment]
Okay. Maybe...

And what goes for "foo" in `window.JST['foo']` = ...? Always the file path minus extension?

**Comment by searls** [comment]
That's what I'd suggest--same path logic as the ES6 ones?

**Comment by joliss** [comment]
The ES6 ones don't have a path at all, they just do `export default` and leave it to the ES6 transpiler (like Esperanto) to figure out the module name based on the path. So we'd be adding path logic.

**Comment by searls** [comment]
Ah, I see. Would that complexity increase be large enough to warrant forking this into a new plugin?

**Comment by joliss** [comment]
Yeah, I'm kinda having trouble figuring out a nice API that would support both global and ES6, so I suspect it might be less confusing to have a separate plugin. We can link it from this repo's README too.

**Comment by joliss** [comment]
(If you want to do this: Feel free to copy everything from this repo, and no need to credit me. It's all trivial code.)
